### PR TITLE
Improve LICENSE and NOTICE files

### PR DESCRIPTION
**Pull request by zhoujinsong**
## Why are the changes needed?

Improve content in LICENSE and NOTICE files.

## Brief change log

- Improve content in LICENSE and NOTICE files.

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by zhoujinsong** [comment]
Amoro copied some code from Apache Iceberg, Apache Paimon, and Apache Flink.

Can you help to review if this PR is okay? @zhouxinyu @justinmclean @yaooqinn @carp84

File: NOTICE [hunk 1]
```diff
@@ -3,3 +3,12 @@ Copyright 2024 The Apache Software Foundation
...
 The Apache Software Foundation (http://www.apache.org/).
+
+Apache Iceberg
+Copyright 2017-2022 The Apache Software Foundation
+
+Apache Paimon
+Copyright 2023-2024 The Apache Software Foundation
+
+Apache Flink
+Copyright 2014-2023 The Apache Software Foundation
```
Comment (yaooqinn) [inline NOTICE:14]:
```suggestion
Copyright 2014-2024 The Apache Software Foundation
```

File: NOTICE [hunk 2]
```diff
@@ -3,3 +3,12 @@ Copyright 2024 The Apache Software Foundation
 
 This product includes software developed at
 The Apache Software Foundation (http://www.apache.org/).
+
+Apache Iceberg
+Copyright 2017-2022 The Apache Software Foundation
```
Comment (yaooqinn) [inline NOTICE:8]:
```suggestion
Copyright 2017-2024 The Apache Software Foundation
```

**Review by yaooqinn — APPROVED** [review]

File: NOTICE (hunk 1 above)
Comment (zhoujinsong) [reply NOTICE:14]: I copied this notice content from https://github.com/apache/iceberg/blob/main/NOTICE.

I am not sure if we should change it.

Comment (zhouxinyu) [reply NOTICE:14]: It's OK to set the correct copyright year. And we can also submit pull requests to correlative projects to fix this issue.

Comment (zhoujinsong) [reply NOTICE]: Thanks for the information, I have changed it.

File: NOTICE (hunk 2 above)
Comment (zhoujinsong) [reply NOTICE]: Fixed.

**Pull request merged**